### ISSUE TITLE
New project name?

### ISSUE THREAD

**Issue by eandersson** [eandersson/amqpstorm#30]
Hey,

So while AMQP-Storm has existed, as well... AMQP-Storm for two years now, and I have been thinking about re-branding the project for a long time. This is primarily because the current name, is easily confused with StormMQ. Instead, I would like something more unique.

The implications of changing the name of a project are huge, but as the library is extremely stable at the moment, and is unlikely to be changed a lot over the next year or so. I think it would give people sufficient time to migrate to the new name.

Anyway, I was hoping to get some feedback on the idea, and hopefully some neat name suggestions!

**Comment by LipillaiDave** [comment]
How about Rabbit-Py? You should contact them. Storm is about 3 orders of magnitude better than Pika, both in usability and practicality.

**Comment by fuzzyami** [comment]
Keeping with the rabbit/bunny/pika terminology: how about LagomorhMQ ? or HyraxMQ?

**Comment by eandersson** [comment]
@LipillaiDave: There is unfortunately already a library called rabbitpy. That kind of name would be ideal, as it is pretty clear that it is a RabbitMQ Client library.

@fuzzyami: I like HyraxMQ. It's got my +1 for now! The only down-side to me is that it sounds like a  service, rather than a library.

**Comment by LipillaiDave** [comment]
What was the rationale for calling it Storm in the first place?

**Comment by eandersson** [comment]
@LipillaiDave: I tried a couple of names, and nothing really clicked. So I practically picked one at random. The whole project wasn't really planned, I was just pissed at the various bugs in Pika at the time, and wanted to see if I could make something myself.

**Comment by ralienpp** [comment]
Long time lurker here. I have used `pika` and struggled with it for quite a while before managing to solve my problem. I then learned about `amqpstorm`, though I did not use it - I keep an eye on all the activity here, to ensure that the project is alive, so I can use it for my next project.

I am happy with what I've seen so far and I must admit that if I knew about this library earlier, I could have had a much more simple structure in my current system. It ended up being quite complex in order to make sure that `pika` fits.  (If you're wondering, it is because I have multiple threads there).

Thus, what brought me here is **thread-safety**. If this is the killer-feature of the library, then perhaps it could be reflected in the name. For example, "thread" is `filum` in Latin - so perhaps it deserves a spot in the name.

"Safe" is `salvus` in Latin, according to Google translate, it also means "free from difficulties".


In other words, I would tackle this problem by making a list of what I want the name to stand for, and then choose one. 

This reasoning might bring us to `salvus-amqp`. You can also try other languages until you stumble upon a nice-sounding word.


I am no big supporter of the rabbit theme, because there are other AMQP servers besides RabbitMQ. I know that it is witty, but then maybe it could be called "HoneyAMQP" or "SweetAMQP" because of HiveMQ.

**Comment by LipillaiDave** [comment]
@Alex: Storm is technically AMQ agnostic but really integrates and works 
best with Rabbit.

As far as thread-safety goes, I have an application that has 10 
different threads (!) with connections through this library and the only 
issue I ever had with it, the maintainer worked with me to fix it 
quickly. As long as I don't do anything too stupid myself, it is very 
stable & works nicely. I've also got a throughput of a few thousands 
messages a second with it, but that is really constrained by my very 
poorly designed DB queries and list processing that I'm doing for each one.

My application originally supported both Pika, this, and other libraries 
sort of like plugins, but I have abandoned the others as they could not 
keep up with increasingly complex requirements.

Anyhow. Point is, I think that the rabbit theme is appropriate because 
of the tight integration with RabbitMQ. It is fast like a rabbit, lean 
like a rabbit, a bunch of them live together like rabbits (comparing 
threads to rabbit warrens), and a RabbitMQ management API may even be in 
the works.

I think they should rename Pika to Amami (a living fossil, endangered, 
primitive rabbit) and Storm to Pika so they don't have to even change 
the docs about what is the recommended library.

But realistically, WarrenMQ sounds good to me. For rabbit warrens and 
thread safety, and it also sounds kind of like "warden."

**Comment by eandersson** [comment]
Thanks everyone for the input. I decided against changing the name for now and instead I got a domain for the project. If anyone is good at Sphinx, or wants to learn, feel free to contribute to the documentation.
https://www.amqpstorm.io/
